src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorNetworkIT.java

```diff
@@ -0,0 +1,359 @@
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.SinkConnectorConfig.TOPICS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class ElasticsearchConnectorNetworkIT extends BaseConnectorIT {
+
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(options().dynamicPort(), false);
+
+  protected static final int NUM_RECORDS = 5;
+  protected static final int TASKS_MAX = 1;
+
+  protected static final String CONNECTOR_NAME = "es-connector";
+  protected static final String TOPIC = "test";
+  protected Map<String, String> props;
+
+  @Before
+  public void setup() {
+    startConnect();
+    connect.kafka().createTopic(TOPIC);
+    props = createProps();
+
+    stubFor(any(anyUrl()).atPriority(10).willReturn(ok()));
+  }
+
+  @After
+  public void cleanup() {
+    stopConnect();
+  }
+
+  /**
+   * Transformer that blocks all incoming requests until {@link #release(int)} is called
+   * to fairly unblock a given number of requests.
+   */
+  public static class ConcurrencyTransformer extends ResponseTransformer {
+
+    private final Semaphore s = new Semaphore(0, true);
+    private final AtomicInteger requestCount = new AtomicInteger();
+
+    @Override
+    public Response transform(Request request, Response response, FileSource files, Parameters parameters) {
+      try {
+        s.acquire();
+      } catch (InterruptedException e) {
+        throw new ConnectException(e);
+      } finally {
+        s.release();
+      }
+      requestCount.incrementAndGet();
+      return response;
+    }
+
+    @Override
+    public String getName() {
+      return "concurrency";
+    }
+
+    public void release(int permits) {
+      s.release(permits);
+    }
+
+    /**
+     * How many requests are currently blocked
+     */
+    public int queueLength() {
+      return s.getQueueLength();
+    }
+
+    /**
+     * How many requests have been processed
+     */
+    public int requestCount() {
+      return requestCount.get();
+    }
+
+    @Override
+    public boolean applyGlobally() {
+      return false;
+    }
+
+  }
+
+  @Test
+  public void testRetry() throws InterruptedException {
+    wireMockRule.stubFor(post(urlPathEqualTo("/_bulk"))
+            .inScenario("bulkRetry1")
+            .whenScenarioStateIs(Scenario.STARTED)
+            .withRequestBody(WireMock.containing("{\"doc_num\":0}"))
+            .willReturn(aResponse().withStatus(500))
+            .willSetStateTo("Failed"));
+
+    wireMockRule.stubFor(post(urlPathEqualTo("/_bulk"))
+            .inScenario("bulkRetry1")
+            .whenScenarioStateIs("Failed")
+            .withRequestBody(WireMock.containing("{\"doc_num\":0}"))
+            .willSetStateTo("Fixed")
+            .willReturn(okJson("{\n" +
+                    "   \"took\": 30,\n" +
```

Comment:
this seems a brittle way to construct a json, should we use JsonObject or is this a pattern among the existing code?